core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -706,10 +709,27 @@ protected File getRootDirFor(String name) {
      * {@link #getActiveInstance()} provides the checked versions of the method.
      * @return The instance. Null if the {@link Jenkins} instance has not been started,
      * or was already shut down
+     * @since 1.653
      */
-    @CLIResolver
     @CheckForNull
+    public static Jenkins getInstanceOrNull() {
+        return HOLDER.getInstance();
+    }
+
+    /**
+     * Gets the {@link Jenkins} singleton. In certain rare cases you may have code that is intended to run before
+     * Jenkins starts or while Jenkins is being shut-down. For those rare cases use {@link #getInstanceOrNull()}.
+     * In other cases you may have code that might end up running on a remote JVM and not on the Jenkins master,
+     * for those cases you really should rewrite your code so that when the {@link Callable} is sent over the remoting
+     * channel it uses a {@code writeReplace} method or similar to ensure that the {@link Jenkins} class is not being
+     * loaded into the remote class loader
```

Comment:
(IOW, I agree that calling `getInstanceOrNull` is the wrong fix for such cases: they should be fixed to not refer to `Jenkins` to begin with. There are some crazy idioms like [this](https://github.com/jenkinsci/subversion-plugin/blob/57460ce3cfb0885fce878c47eafec45e83e2fc00/src/main/java/hudson/scm/SubversionWorkspaceSelector.java#L105-L108) that caused headaches for SECURITY-144 anyway.)